hahomematic/platforms/custom/cover.py

```diff
@@ -279,22 +294,56 @@ async def _set_level(
         collector: CallParameterCollector | None = None,
     ) -> None:
         """
-        Move the cover to a specific tilt level. Value range is 0.0 to 1.01.
+        Move the cover to a specific tilt level. Value range is 0.0 to 1.00.
 
-        1.01 means no change.
+        level or tilt_level may be set to None for no change.
         """
-        _level = level if level is not None else self.current_position / 100.0
-        _tilt_level = tilt_level if tilt_level is not None else self.current_tilt_position / 100.0
+        currently_moving = False
+
+        async with self._command_processing_lock:
+            if level is not None:
+                _level = level
+            elif self._target_level is not None:
+                # The blind moves and the target blind height is known
+                currently_moving = True
+                _level = self._target_level
+            else:  # The blind is at a standstill and no level is explicitly requested => we remain at the current level
+                _level = self._channel_level
+
+            if tilt_level is not None:
+                _tilt_level = tilt_level
+            elif self._target_tilt_level is not None:
+                # The blind moves and the target slat position is known
+                currently_moving = True
+                _tilt_level = self._target_tilt_level
+            else:  # The blind is at a standstill and no tilt is explicitly desired => we remain at the current angle
+                _tilt_level = self._channel_tilt_level
+
+            if currently_moving:
+                # Blind actors are buggy when sending new coordinates while they are moving. So we stop them first.
+                await self._stop()
+
+            await self._transmit_position(
+                level=_level, tilt_level=_tilt_level, collector=collector
+            )
+
+    @bind_collector(use_command_queue=False)
```

Comment:
I moved the collectors further "down" the call graph. The reason is that the `_target` properties should be changed before the lock is released, so that no race conditions can occur.